fabric-command-api-v1/src/main/java/net/fabricmc/fabric/api/client/command/v1/FabricClientCommandSource.java

```diff
@@ -56,12 +60,49 @@
 	 *
 	 * @return the player
 	 */
-	ClientPlayerEntity getPlayer();
+	AbstractClientPlayerEntity getPlayer();
```

Comment:
Hmm, this change may not be backward compatible...

I suggest declare another
```java
default AbstractClientPlayerEntity getTargetPlayer() {
    return getPlayer();
}
```

and possibly deprecate (or not) the existing `getPlayer()` as well.

Reply:
```suggestion
	ClientPlayerEntity getPlayer();
```
for binary compatibility (otherwise existing code calling `FabricClientCommandSource.getPlayer` expecting a `ClientPlayerEntity` return would get a no such method error)

Reply:
[This](https://github.com/FabricMC/fabric/pull/1518#discussion_r669211123) does not allow other players though, as suggested [here](https://github.com/FabricMC/fabric/pull/1518#issuecomment-867296548) by you. Do you want me to remove that feature?

Reply:
Sorry, I am dumb! Player says that we can just use `getEntity` instead, so we proably will revert changes to this and remove `getTargetPlayer`; instead we will document this `getPlayer` simply returns `getClient().player` than the actual entity that executes the command, and ask users to do instanceof checks on the `getEntity` result.

Reply:
Sure!